File: leetcode/00013. Roman to Integer/C++/sol.cpp
// Hashmap Solution
class Solution {
public:
    int romanToInt(string s) {
        unordered_map<char, int> map = {{'I',1},{'V',5},{'X',10},{'L',50},{'C',100},{'D',500},{'M',1000}};
        int num = 0;
        for (int i = 0; i < s.length()-1; i++)
        {
            int now = map[s[i]];
            int then = map[s[i+1]];
            num += now >= then ? now : -now;
        }
        return num+map[s[s.length()-1]];
    }
};

File: leetcode/00013. Roman to Integer/Python/sol.py
# Dictionary Solution
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        map = {'I':1, 'V':5, 'X':10, 'L':50, 'C':100, 'D': 500, 'M':1000}
        num = 0
        l = len(s)-1
        for i in range(l):
            if map[s[i]] < map[s[i+1]]:
                num += -map[s[i]]
            else:
                num += map[s[i]]
        return num+map[s[-1]]